opentelemetry-sdk/src/metrics/meter.rs

```diff
@@ -479,6 +513,75 @@ impl InstrumentProvider for Meter {
     }
 }
 
+/// Validation policy for instrument
+#[derive(Clone, Copy)]
+enum InstValidationPolicy {
+    HandleGlobalAndIgnore,
+    /// Currently only for test
+    #[cfg(test)]
+    Strict,
+}
+
+fn validate_instrument_config(
+    name: &str,
+    unit: Option<&Unit>,
+    policy: InstValidationPolicy,
+) -> Result<()> {
+    match validate_instrument_name(name).and_then(|_| validate_instrument_unit(unit)) {
+        Ok(_) => Ok(()),
+        Err(err) => match policy {
+            InstValidationPolicy::HandleGlobalAndIgnore => {
+                global::handle_error(err);
+                Ok(())
+            }
+            #[cfg(test)]
+            InstValidationPolicy::Strict => Err(err),
+        },
+    }
+}
+
+fn validate_instrument_name(name: &str) -> Result<()> {
+    if name.is_empty() {
+        return Err(MetricsError::InvalidInstrumentConfiguration(
+            INSTRUMENT_NAME_EMPTY,
+        ));
+    }
+    if name.len() > 255 {
+        return Err(MetricsError::InvalidInstrumentConfiguration(
+            INSTRUMENT_NAME_LENGTH,
+        ));
```

Comment:
Why prefer a hardcoded `255` and a constant for the error string over a constant for the threshold value defined in the spec and a literal for the string?

Reply:
thanks! there is an `InstProvider`, so I tried to be consistent. but rename to `InstrumentValidationPolicy`